pal/input/web/accelerometer-input.ts

```diff
@@ -120,12 +121,13 @@ export class AccelerometerInputSource {
 
     public start (): void {
         // for iOS 13+, safari
-        if (window.DeviceMotionEvent && typeof DeviceMotionEvent.requestPermission === 'function') {
-            DeviceMotionEvent.requestPermission().then((response) => {
+        // NOTE: since TS 4.4, requestPermission is not defined in class DeviceMotionEvent, this should be a breaking change in TS.
+        if (window.DeviceMotionEvent && typeof (DeviceMotionEvent as any).requestPermission === 'function') {
```

Comment:
What does the comment mean? As it just convert it to `any`.

Reply:
that means there is a breaking change between TS 4.3 and TS 4.4

DeviceMotionEvent is defined differently in both version, so we can only convert DeviceMotionEvent as any to access the requestPermission property

Reply:
If it is not defined in `DeviceMotionEvent `, just make it as `any` can fix the problem?

Reply:
yes, `requestPermission ` is a safari-specific method of DeviceMotionEvent, not a web standard method, so it's not defined in TS standard dom lib since 4.4

as a cross-platform engine, we still need to keep this broswer compabiliry

Reply:
see https://caniuse.com/mdn-api_devicemotionevent_requestpermission_static

Reply:
What i confused is the comment. Why need it?

Reply:
I updated the comment, maybe this would help understanding 
https://github.com/cocos/cocos-engine/pull/15899/commits/99cb40dc250b0c76f5bd712357a084d3ff66468a

Reply:
Yep. It is more clear. But how do you know it is a bug of `lib.dom.d.ts` or it is already removed in the specification, and some web platforms may remove the function.

Reply:
here is the release note of TS 4.4, the breaking change list here https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#libdts-changes-for-typescript-44

> some web platforms may remove the function

the declaration of `requestPermission` is only removed in TS 4.4 `lib.dom.d.ts `.   
and some web platforms actually didn't remove it, they just never support this interface, so our implementation here need to check whether `requestPermission` is defined before we use it